src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20220608_194500.java

```diff
@@ -0,0 +1,119 @@
+package de.tum.in.www1.artemis.config.migration.entries;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Lists;
+
+import de.tum.in.www1.artemis.config.migration.MigrationEntry;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.repository.ResultRepository;
+
+/**
+ * This migration extracts the information about a programming result and stores it inside the new variables so the result String can still be displayed correctly in the client
+ */
+@Component
+public class MigrationEntry20220608_194500 extends MigrationEntry {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MigrationEntry20220608_194500.class);
+
+    private final ResultRepository resultRepository;
+
+    public MigrationEntry20220608_194500(ResultRepository resultRepository) {
+        this.resultRepository = resultRepository;
+    }
+
+    /**
+     * Retrieves all results from the database and executes the processing method in batches of 100 results to prevent database timeouts
+     */
+    @Override
+    public void execute() {
+        List<Result> results = resultRepository.findAll();
+        results.removeIf(result -> !(result.getParticipation() instanceof ProgrammingExerciseParticipation));
+
+        List<Result> unprocessedResults = new ArrayList<>();
+        List<Result> processedResults = new ArrayList<>();
+        results.forEach(result -> {
+            if (result.getResultString() != null && result.getResultString().matches(".*of.*passed.*")) {
+                unprocessedResults.add(result);
+            }
+            else {
+                processedResults.add(result);
+            }
+        });
+
+        LOGGER.info("Found {} results to process.", unprocessedResults.size());
+
+        Lists.partition(results, 100).forEach(resultList -> {
```

Comment:
```suggestion
        Lists.partition(unprocessedResults, 100).forEach(resultList -> {
```
Is that what you want?